/*
 * If not stated otherwise in this file or this component's LICENSE file the
 * following copyright and licenses apply:
 *
 * Copyright 2022 Sky UK
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto2";

import "google/protobuf/descriptor.proto";
import "rialtoipc.proto";

package firebolt.rialto;

// You need this to generate the rpc service stubs
option cc_generic_services = true;

message TestEventSingleVar {
    required int32      var1 = 1;
}

message TestEventMultiVar {
    enum TestType {
        ENUM1 = 0;
        ENUM2 = 1;
    }
    required int32      var1 = 1;
    required uint32     var2 = 2;
    required TestType   var3 = 3;
    required string     var4 = 4;
}

message TestNoVar {
}
message TestSingleVar {
    required int32 var1 = 1;
}
message TestMultiVar {
    enum TestType {
        ENUM1 = 0;
        ENUM2 = 1;
    }
    required int32      var1 = 1;
    required uint32     var2 = 2;
    required TestType   var3 = 3;
    required string     var4 = 4;
}

message TestSingleVarNoReply {
    required int32 var1 = 1;
}

service TestModule {
    rpc TestRequestSingleVar(TestSingleVar) returns (TestNoVar) {
    }
    rpc TestRequestMultiVar(TestMultiVar) returns (TestNoVar) {
    }
    rpc TestResponseSingleVar(TestNoVar) returns (TestSingleVar) {
    }
    rpc TestResponseMultiVar(TestNoVar) returns (TestMultiVar) {
    }
    rpc TestRequestSingleVarNoReply(TestSingleVarNoReply) returns (TestNoVar) {
        option (rialto.ipc.no_reply) = true;
    }
}
